apps/aecore/src/aec_chain_state.erl

```diff
@@ -42,6 +43,22 @@ get_n_key_headers_backward_from(Header, N) ->
     Node = wrap_header(Header),
     get_n_key_headers_from(Node, N).
 
+%% Compute the median timestamp for last aec_governance:median_timestamp_key_blocks()
+-spec median_timestamp(aec_headers:header()) -> {ok, integer()} | 'error'.
+median_timestamp(Header) ->
+    TimeStampKeyBlocks = aec_governance:median_timestamp_key_blocks(),
+    case aec_headers:height(Header) =< TimeStampKeyBlocks of
+        true ->
+            {ok, ?GENESIS_TIME};
+        false ->
+            case get_n_key_headers_from(wrap_header(Header), TimeStampKeyBlocks + 1) of
```

Comment:
If e.g. Header is header of key block and has height 12, then get_n_key_headers_from is called with the (wrapped) Header and 12 (11+1).
Then get_n_key_headers_from returns list of headers - exactly 11 - where the last is header at height 12 itself and first (list head) is header at height 2 (note: header at height 1 is missing). Then line below takes tail (`tl`) - that keeps headers from 3 to 12 (note: 12 is kept in computation of median for verification of 12 itself).

I feel I may be reading this code incorrectly.

Reply:
I see tl replaced with lists:droplast - ok.

Reply:
Yes, well spotted.